clang/include/clang/Sema/SemaSYCL.h

```diff
@@ -265,6 +265,8 @@ class SemaSYCL : public SemaBase {
 
   llvm::DenseSet<const FunctionDecl *> SYCLKernelFunctions;
 
+  llvm::SmallVector<FunctionDecl *, 8> FreeFunctionDeclarations;
```

Comment:
I would maybe use a Set container to make sure all memorized declarations are unique.